src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -676,6 +665,36 @@ private static bool IsPathRootedSafe(string path)
             // Otherwise, Path.IsPathRooted() will throw an ArgumentException.
             return path.IndexOfAny(Path.GetInvalidPathChars()) == -1 && Path.IsPathRooted(path);
         }
+
+        private IReadOnlyList<PackageSource> GetAllPackageSources()
+        {
+            return NuGetUIThreadHelper.JoinableTaskFactory.Run(async () => { return await GetAllPackageSourcesAsync(); });
+        }
+
+        private async Task<IReadOnlyList<PackageSource>> GetAllPackageSourcesAsync()
+        {
+            var remoteBroker = await BrokeredServicesUtilities.GetRemoteServiceBroker();
+            var nuGetSettingsService = await remoteBroker.GetProxyAsync<INuGetSourceRepositoryService>(NuGetBrokeredServices.SourceRepositoryProviderService);
+            using (nuGetSettingsService as IDisposable)
+            {
+                return await nuGetSettingsService.GetAllPackageSourcesAsync(CancellationToken.None);
+            }
+        }
+
+        private void SaveAllPackageSources(IReadOnlyList<PackageSource> packageSources)
+        {
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () => await SaveAllPackageSourcesAsync(packageSources));
+        }
+
+        private async Task SaveAllPackageSourcesAsync(IReadOnlyList<PackageSource> packageSources)
+        {
+            var remoteBroker = await BrokeredServicesUtilities.GetRemoteServiceBroker();
+            var nuGetSettingsService = await remoteBroker.GetProxyAsync<INuGetSourceRepositoryService>(NuGetBrokeredServices.SourceRepositoryProviderService);
+            using (nuGetSettingsService as IDisposable)
+            {
+                await nuGetSettingsService.SaveAllPackageSourcesAsync(packageSources, CancellationToken.None);
```

Comment:
Check for null here again.
Tip: use `LangVersion=8` and add `#nullable enable` at the top of your file (or at least as regions around your new code) so that C# can help you avoid missing these null checks.